C++/include/algorithm/dynamic_programming/longest_increasing_subsequence.hpp

```diff
@@ -0,0 +1,47 @@
+
+/*
+  Longest Increasing Subsequence
+  ------------------------------
+  Given an unsorted array of integers, return the longest increasing subsequence length
+
+  Time complexity
+  --------------
+  O(N^2), where N is size of array
+
+  Space complexity
+  ---------------
+  O(N), where N is size of array
+
+  Author
+  ------
+  Shikhar Jaiswal(@shikherjaiswal)
+*/
+#ifndef longest_increasing_subsequence_hpp
+#define longest_increasing_subsequence_hpp
+#include <vector>
+
+// lis function return longest increasing subsequence length given a vector and its size 'n'
```

Comment:
This is more what I was thinking in terms of function comments: https://github.com/faheel/Algos/blob/reorganise/C%2B%2B/include/algorithm/backtracking/n_queens.hpp